pkg/agent/openflow/pod_connectivity.go

```diff
@@ -152,16 +158,105 @@ func (f *featurePodConnectivity) initFlows() []binding.Flow {
+	var flows []binding.Flow
+	for _, port := range srcOfPorts {
+		if direction == v1alpha2.DirectionIn || direction == v1alpha2.DirectionBoth {
+			flows = append(flows, IngressTrafficControlMarkTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchRegFieldWithValue(TargetOFPortField, port).
+				Action().LoadRegMark(trafficControlRegMark).
+				Action().NextTable().
+				Done())
+		}
+		if direction == v1alpha2.DirectionOut || direction == v1alpha2.DirectionBoth {
+			flows = append(flows, EgressTrafficControlMarkTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchInPort(port).
+				Action().LoadRegMark(trafficControlRegMark).
+				Action().NextTable().
+				Done())
+		}
+	}
+	return flows
+}
+
+// trafficControlInitFlows generates the init flows for a given mark value.
+func (f *featurePodConnectivity) trafficControlInitFlows(mark uint32, action v1alpha2.Action, dstOfInPort, dstOfOutPort uint32) []binding.Flow {
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	var flows []binding.Flow
+	trafficControlCTMark := binding.NewCTMark(ConnTrafficControlCTMarkField, mark)
+	if action == v1alpha2.ActionMirror {
+		// When the given mark is for action mirror, this generates the flow to output packets to the original port as
+		// well as mirror the packets to another given port in L2ForwardingOutTable.
+		flows = append(flows, L2ForwardingOutTable.ofTable.BuildFlow(priorityHigh+1).
+			Cookie(cookieID).
+			MatchCTMark(trafficControlCTMark).
+			MatchRegMark(OFPortFoundRegMark).
+			Action().OutputToRegField(TargetOFPortField).
+			Action().Output(dstOfInPort).
+			Done())
+	} else if action == v1alpha2.ActionRedirect {
+		flows = append(flows,
+			// When the given mark is for action redirect, the generates the flow to forward the redirected packets sourced
+			// from a given port to stageSwitching directly in ClassifierTable.
+			ClassifierTable.ofTable.BuildFlow(priorityHigh).
+				Cookie(cookieID).
+				MatchInPort(dstOfOutPort).
+				Action().LoadRegMark(FromRedirectRegMark).
+				Action().GotoStage(stageSwitching).
+				Done(),
+			// When the given mark is for action redirect, the generates the flow to forward the packets to be redirected
+			// (with CT mark whose value is equal to the given mark value) to the given port.
+			L2ForwardingOutTable.ofTable.BuildFlow(priorityHigh+1).
+				Cookie(cookieID).
+				MatchCTMark(trafficControlCTMark).
+				MatchRegMark(OFPortFoundRegMark).
+				Action().Output(dstOfInPort).
+				Done(),
+		)
+	}
+	return flows
+}
+
+// redirectedPktsFwdFlows generates the flows to forward the redirected packets.
+func (f *featurePodConnectivity) redirectedPktsFwdFlows() []binding.Flow {
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	return []binding.Flow{
+		// This generates the flow to forward the redirected packets after loading output port to reg1 in L2ForwardingCalcTable
+		// to stageOutput directly in TrafficControlRedirectOutTable.
```

Comment:
Better to move "after loading output port to reg1 in L2ForwardingCalcTable" to be end?

> This generates the flow to forward the redirected packets to stageOutput directly in TrafficControlRedirectOutTable, after loading output port to reg1 in L2ForwardingCalcTable.

Reply:
Updated, thanks.